river/neighbors/knn_classifier.py

```diff
@@ -1,176 +1,193 @@
-import typing
+from river import base, utils
 
-from river import base
-from river.utils import dict2numpy
-from river.utils.math import softmax
+from .base_neighbors import BaseKNN
+from .neighbors import DistanceFunc
 
-from .base_neighbors import BaseNeighbors
+__all__ = ["KNNClassifier"]
 
 
-class KNNClassifier(BaseNeighbors, base.Classifier):
-    """k-Nearest Neighbors classifier.
-
-    This non-parametric classification method keeps track of the last
-    `window_size` training samples. The predicted class-label for a
-    given query sample is obtained in two steps:
+class KNNClassifier(BaseKNN, base.Classifier):
+    """
```

Comment:
The `K-Nearest Neighbors (KNN) for classification.` should be placed here. Take any other River class as an example for the docstring format